AzureMonitorAgent/agent.py

```diff
@@ -1146,8 +1146,22 @@ def find_package_manager(operation):
     for rpm_dist in rpm_set:
         if dist.startswith(rpm_dist):
             PackageManager = "rpm"
-            # Same as above.
+            # Same as above; need to force.
             PackageManagerOptions = "--force"
+
+            # If FIPS is enabled and RPM version is 4.14 or higher, we need to apply the below workaround
+            try:
+                fips_code, _ = run_command_and_log('grep -i "fips=1" /proc/cmdline 1>/dev/null 2>&1', check_error=False, log_cmd=False)
+                if fips_code == 0:
+                    rpm_code, rpm_output = run_command_and_log('rpm --version', check_error=False, log_cmd=False)
+                    if rpm_code == 0:
+                        rpm_version = (rpm_output.split(" ")[-1]).split(".")
```

Comment:
If fips is enabled on rpm where version is smaller than 4.14, digest flag will be still needed to install the package. I think you can just remove the version check entirely. What are you trying to accomplish with the check?

Reply:
Per my understanding based on RPM forum posts and my personal testing RPM <4.14 does _not_ verify the digests, so we don't need the flags in that case. We should make scope of the flag as narrow as possible.

Reply:
I still think that is less hassle and less time for installer if we skip digest validation just based on the enable flag. but signed off since you confirmed that with fips enabled rpm 4.14 do not check the digest during install.